### PR TITLE
fix(toolbar): add missing focus trigger when right key press and last button focused FE-2823

### PR DESCRIPTION
**Pull request by edleeks87**
### Proposed behaviour

Creates a new `LinkPreview` component to render data relating to links added in components like the
`TextEditor` as a `div` element. They will also display in the `Note` component as a read only `a`
element. The component has a loading state controlled by the `isLoading` prop and supports passing
in a `title`, a `description` and an `image` config object along with the url for the associated
`EditorLink`. If no `image` config is passed a placeholder image is used.

Adds `previews` and `onLinkAdded` props to `TextEditor` to support the rendering of
`LinkPreview` for any urls that have associated meta data. The `onLinkAdded` callback is
integrated with a context to allow `EditorLink` components to report when they have been
rendered via the decorator.

Links in Note can now be focused via tabbing

Adds `preview` prop to support rendering `LinkPreview` beneath the note content

### Current behaviour

Keyboard navigation in Toolbar controls does not work correctly, cannot focus first control via pressing right key until it cycles round for example.

No `EditorLinkPreview` or support for rendering them in either `TextEditor` or `Note` components.

Links in Note cannot be focused via tabbing

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context

image

image

image

image

### Testing instructions

`design-system-text-editor--with-link-previews` and `design-system-note--with-previews` added to storybook.

https://codesandbox.io/s/link-preview-demo-iz934
https://codesandbox.io/s/demo-previews-with-editor-building-previews-xw52n
https://codesandbox.io/s/demo-previews-with-note-fetching-ijvsj

### CODE REVIEW

File: src/components/text-editor/text-editor.stories.mdx [hunk 1]
```diff
@@ -261,6 +262,62 @@ which is multiplied by the line-height (21px).
   </Story>
 </Preview>
 
+### With link previews
+It is possible to render to render `EditorLinkPreview`s via the `previews` prop. The `onLinkAdded` prop provides a callback
```
Comment (dawidzarzycki) [inline src/components/text-editor/text-editor.stories.mdx:266]: `to render` twice

File: src/components/link-preview/link-preview.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
+
+import LinkPreview from '.';
+
+<Meta
+  title="Design System/Link Preview"
+  parameters={{ info: { disable: true }}}
+/>
+
+# Message
```
Comment (DipperTheDan) [inline src/components/link-preview/link-preview.stories.mdx:10]: Should this be `Message`? The component is called `Link Preview`.

File: src/components/link-preview/link-preview.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      title="This is an example of a title"
+      url="https://www.sage.com"
+      description="Captain, why are we out here chasing comets? I'd like to think that I haven't changed those things, sir. Computer, lights up! Not if I weaken first. Damage report! Yesterday I did not know how to eat gagh. The Federation's gone; the Borg is everywhere! We know you're dealing in stolen ore. But I wanna talk about the assassination attempt on Lieutenant Worf. Our neural pathways have become accustomed to your sensory input patterns. Wouldn't that bring about chaos?"
+    />
+  </Story>
+</Preview>
+
+## Props
+
+### Message
```
Comment (DipperTheDan) [inline src/components/link-preview/link-preview.stories.mdx:69]: Same again. I thought this should be `Link Preview`?

File: src/components/link-preview/link-preview.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+If no `url` is passed or the `isLoading` prop is true the component will display in a loading state. When in this state, it 
+renders as a `div` element: it cannot be focused via tabbing and clicking will not trigger anything.
+
+<Preview>
+  <Story name="with loading state">
+    <LinkPreview isLoading />
+  </Story>
+</Preview>
+
+### With close icon
```
Comment (DipperTheDan) [inline src/components/link-preview/link-preview.stories.mdx:50]: Axe is throwing an accessibility error for this example. 
`Ensures buttons have discernible text`

An `aria-label` for the close icon should resolve this violation.

File: src/components/link-preview/link-preview.stories.mdx (hunk 2 above)
Comment (edleeks87) [reply src/components/link-preview/link-preview.stories.mdx:10]: Copy and paste 🤦

**Review by dawidzarzycki — DISMISSED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by dawidzarzycki — DISMISSED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by dawidzarzycki — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by dawidzarzycki — DISMISSED** [review]

**Review by sianford — APPROVED** [review]

**Comment by sianford** [comment]
Tested in Safari, Chrome, Firefox and Edge in Storybook and codesandbox. Testing passed ✅

**Review by dawidzarzycki — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**